OJS.Workers.ExecutionStrategies/Java/JavaSpringAndHibernateProjectExecutionStrategy.cs

```diff
@@ -501,5 +429,39 @@ private string ExtractEntryPointFromPomXml(string submissionFilePath)
             FileHelpers.DeleteFiles(pomXmlPath);
             return packageName.InnerText.Trim();
         }
+
+        private string GetMavenErrorsComment(string testOutput)
+        {
+            var sb = new StringBuilder();
+
+            foreach (var line in testOutput.Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries)
+                         .Where(x => x
+                             .StartsWith("[ERROR]") || x.StartsWith("[FAILURE]")))
+            {
+                sb.Append("\t" + line);
+            }
+
+            return sb.ToString();
+        }
+
+        private string EvaluateMavenTestOutput(string testOutput, Regex testErrorMatcher)
+        {
+            var message = TestPassedMessage;
+            var errorMatch = testErrorMatcher.Match(testOutput);
+
+            if (!errorMatch.Success)
+            {
+                return message;
+            }
+
+            return this.GetMavenErrorsComment(testOutput);
+        }
+
+        private bool ValidateFolderStructure(string submissionFilePath)
+        {
+            var paths = FileHelpers.GetFilePathsFromZip(submissionFilePath).ToList();
+
+            return paths.Any(x => x.StartsWith(MainCodeFolderPattern)) && paths.Any(x => x.StartsWith(PomXmlFileNameAndExtension));
+        }
```

Comment:
Shouldn't `.All` be used here like:
`paths.All(x => x.StartsWith(MainCodeFolderPattern) || x.StartsWith(PomXmlFileNameAndExtension))`.
Do we assume a structure is valid if there are any other paths than these?

Reply:
Yes, we are not limiting users to not having different paths. They can add resources or something else in the solution that we do not care about and the whole thing that the validation should do here is to validate that there is a file /pom.xml and that there are files for compiling in the main code folder.